### PR TITLE
ocs-to-ocs: fix all the panic

### PR DESCRIPTION
**Pull request by subhamkrai**
fixing all the panic nil pointer error

Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

**Comment by subhamkrai** [comment]
PTAL @nb-ohad  @Madhu-1 @umangachapagain @iamniting @rexagod

File: controllers/storageclassclaim/storageclassclaim_controller.go [hunk 1]
```diff
@@ -384,6 +388,7 @@ func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Resul
 		r.cephBlockPool = &rookCephv1.CephBlockPool{}
 		r.cephBlockPool.Name = fmt.Sprintf("cephblockpool-%s", r.storageConsumer.Name)
 		r.cephBlockPool.Namespace = r.namespace
+		r.cephBlockPool.Labels = make(map[string]string)
 		r.cephBlockPool.Labels[controllers.StorageConsumerNameLabel] = r.storageConsumer.Name
```
Comment (nb-ohad) [inline controllers/storageclassclaim/storageclassclaim_controller.go:392]:
```suggestion
		r.cephBlockPool.Labels = map[string]string {
			controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
		}
```

File: controllers/storageclassclaim/storageclassclaim_controller.go [hunk 2]
```diff
@@ -400,6 +405,8 @@ func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Resul
 	r.cephClientNode.Name = controllers.GenerateHashForCephClient(r.storageClassClaim.Name, "node")
 	r.cephClientNode.Namespace = r.namespace
 
+	r.cephResourcesByName = make(map[string]*v1alpha1.CephResourcesSpec)
+
```
Comment (nb-ohad) [inline controllers/storageclassclaim/storageclassclaim_controller.go:409]:
```suggestion
	r.cephResourcesByName = map[string]*v1alpha1.CephResourcesSpec{}
```

File: controllers/storageclassclaim/storageclassclaim_controller.go [hunk 3]
```diff
@@ -366,10 +367,13 @@ func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Resul
 	r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimInitializing
 
 	// reading storageConsumer Name from storageClassClaim ownerReferences
+	r.storageConsumer = &v1alpha1.StorageConsumer{}
 	ownerRefs := r.storageClassClaim.GetOwnerReferences()
 	for i := range ownerRefs {
-		if ownerRefs[i].Kind == r.storageConsumer.Kind {
+		if ownerRefs[i].Kind == "StorageConsumer" {
 			r.storageConsumer.Name = ownerRefs[i].Name
+			r.storageConsumer.Namespace = r.storageClassClaim.Namespace
```
Comment (nb-ohad) [inline controllers/storageclassclaim/storageclassclaim_controller.go:375]: Don't assume same namespace for Claim (user accesiable API)  and consumer (internal API), use storage clsuter instead (which is already loaded):
```suggestion
			r.storageConsumer = &v1alpha1.StorageConsumer{}
			r.storageConsumer.Name = ownerRefs[i].Name
			r.storageConsumer.Namespace = r.storageCluster.Namespace
```

File: controllers/storageclassclaim/storageclassclaim_controller.go [hunk 4]
```diff
@@ -366,10 +367,13 @@ func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Resul
 	r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimInitializing
 
 	// reading storageConsumer Name from storageClassClaim ownerReferences
+	r.storageConsumer = &v1alpha1.StorageConsumer{}
```
Comment (nb-ohad) [inline controllers/storageclassclaim/storageclassclaim_controller.go:370]: Please create the object only if you found an owner ref
```suggestion
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

**Pull request merged**